Test library errors setting headers

Trying the new test library and getting an error trying to set a header 

```
go test ./... -v
=== RUN   TestPlugin
    test.go:571: Access
panic: assignment to entry in nil map

goroutine 19 [running]:
net/textproto.MIMEHeader.Set(...)
        /usr/local/Cellar/go/1.16.4/libexec/src/net/textproto/header.go:22
net/http.Header.Set(...)
        /usr/local/Cellar/go/1.16.4/libexec/src/net/http/header.go:37
github.com/Kong/go-pdk/test.(*TestEnv).Handle(0xc0000f6000, 0xc0000b00d8, 0x18, 0xc00020c1c0, 0xc, 0xc, 0x0, 0x2b2d37, 0x1792440)
        [redacted]/vendor/github.com/Kong/go-pdk/test/test.go:409 +0x209a
github.com/Kong/go-pdk/bridge/bridgetest.MockFunc.func1(0x1529910, 0xc0000da180, 0x15212f8, 0xc0000f6000)
        [redacted]/vendor/github.com/Kong/go-pdk/bridge/bridgetest/bridgetest.go:132 +0xea
created by github.com/Kong/go-pdk/bridge/bridgetest.MockFunc
        [redacted]/vendor/github.com/Kong/go-pdk/bridge/bridgetest/bridgetest.go:117 +0x65
FAIL    plugintest   0.168s
FAIL
```

here is the code

```
// main.go
package main

import (
	"log"

	"github.com/Kong/go-pdk"
	"github.com/Kong/go-pdk/server"
)

func main() {
	server.StartServer(New, Version, Priority)
}

var Version = "0.0.1"
var Priority = 1

type Config struct{}

func New() interface{} {
	return &Config{}
}

func (conf Config) Access(kong *pdk.PDK) {
	err := kong.Response.SetHeader("foo", "bar")
	if err != nil {
		log.Printf("Error getting setting header: %s", err.Error())
	}
}
```

```
// main_test.go
package main

import (
	"testing"

	"github.com/Kong/go-pdk/test"
	"github.com/stretchr/testify/assert"
)
func TestPlugin(t *testing.T) {
	env, err := test.New(t, test.Request{
		Method:  "GET",
		Url:     "http://example.com?q=search&x=9",
	})
	if assert.NoError(t, err) {
		env.DoHttps(&Config{})
		assert.Equal(t, 200, env.ClientRes.Status)
		assert.Equal(t, "bar", env.ClientRes.Headers.Get("foo"))
	}
}
```
```
// go.mod
module plugintest

go 1.16

require (
	github.com/Kong/go-pdk v0.7.0-beta1.0.20210526164613-49a95960d27c
	github.com/stretchr/testify v1.5.1
)
```

--- a/test/test.go
+++ b/test/test.go
@@ -217,8 +217,8 @@ func New(t *testing.T, req Request) (env *TestEnv, err error) {
 		state:      running,
 		ClientReq:  req,
 		ServiceReq: req.clone(),
-		ServiceRes: Response{},
-		ClientRes:  Response{},
+		ServiceRes: Response{Headers: make(http.Header)},
+		ClientRes:  Response{Headers: make(http.Header)},
 	}
 
 	b := bridge.New(bridgetest.MockFunc(env)) // check

